models/marts/core/int_all_graph_resources.sql

```diff
@@ -1,4 +1,17 @@
 -- one row for each resource in the graph
+
+{# flatten the sets of permissable primary key test sets to one level for later iteration #}
+{%- set test_macro_list = [] %}
+{%- for test_set in var('primary_key_test_macros') -%}
+    {%- if test_set is iterable and (test_set is not string and test_set is not mapping) -%}
```

Comment:
Rather than needing to have different behavior for when the test set is a string vs. a list, I wonder if we should just require that each input for the primary_key_test_macros is a list. So the default would be:

`[["dbt.test_unique", "dbt.test_not_null"], ["dbt_utils.test_unique_combination_of_columns"]]`

and the override example would be:

`primary_key_test_macros: [["dbt_constraints.test_primary_key"]]`

Maybe that's worse because it requires you to remember that each set needs [] but would make this piece of code less complex.